hrjobcontract/CRM/Hrjobcontract/BAO/HRJobContractRevision.php

```diff
@@ -210,4 +236,84 @@ protected static function clearEffectiveEndDate($revisionId) {
     );
     CRM_Core_DAO::executeQuery($clearEffectiveEndDateQuery, $clearEffectiveEndDateParams);
   }
+
+  /**
+   * Builds the SQL query for fetching the revision details of the given entities
+   *
+   * The fields of each entities are dynamically listed in the SELECT clause
+   * and are aliased with a prefix ({entity}__) so that it is possible to know
+   * which entity they belong to (civicrm_hrjobcontract_pay.id -> pay__id)
+   *
+   * @param Array $entities
+   * @param Array $revision
+   */
+  protected static function buildFullDetailsQuery($entities, $revision) {
+    $query = ['select' => [], 'from' => [], 'where' => []];
+
+    /**
+     * Creates an alias for the given entity
+     *
+     * @param string $entity
+     * @return  string
+     */
+    $entityAlias = function ($entity) {
```

Comment:
I'm ok with that. I've done it myself a few time on another project. I would just ask to remove the trailing semicolon after the closing bracket